### PR TITLE
Merge PQ bucket into shard store, migrate PQ files from flat file structure

### PR DESCRIPTION
**Pull request by parkerduckworth**
### What's being changed:

Previously, restoring a backed up PQ-enabled class resulted in the loss of all vectors. This is because the shard was unaware of the the LSM store used to house the compressed vectors, so when an index received a backup command, the compressed vector files were ignored.

This was fixed with #3765, but only for pre-1.22 instances whose datafiles are still using the old flat file structure. (See #3719)

This PR makes utilizes the parent shard's LSM store to create a PQ bucket, so now all the correct PQ-related files are included in the backup, fixing backups for PQ-enabled classes using hierarchical file structure.

Additionally, this PR updates the file hierarchy migration logic to include existing PQ vector files

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Review by aliszka — APPROVED** [review]

File: adapters/repos/db/helpers/helpers.go [hunk 1]
```diff
@@ -20,17 +20,10 @@ import (
 var (
 	ObjectsBucket              = []byte("objects")
 	ObjectsBucketLSM           = "objects"
-	CompressedObjectsBucketLSM = "compressed_objects"
+	CompressedVectorsBucketLSM = "vectors"
```
Comment (aliszka) [inline adapters/repos/db/helpers/helpers.go:23]: can it be renamed to `compressed_vectors`? flat index can store uncompressed vectors as well, there will be `VectorsBucketLSM` needed.

**Review by aliszka — APPROVED** [review]

File: adapters/repos/db/helpers/helpers.go [hunk 2]
```diff
@@ -18,21 +18,14 @@ import (
...
-	ObjectsBucket              = []byte("objects")
-	ObjectsBucketLSM           = "objects"
-	CompressedObjectsBucketLSM = "compressed_objects"
-	VectorsFlatBucketLSM       = "vectors_flat"
-	VectorsFlatBQBucketLSM     = "vectors_flat_bq"
-	DimensionsBucketLSM        = "dimensions"
-	DocIDBucket                = []byte("doc_ids")
+	ObjectsBucket          = []byte("objects")
+	ObjectsBucketLSM       = "objects"
+	VectorsPQBucketLSM     = "vectors_pq"
```
Comment (aliszka) [inline adapters/repos/db/helpers/helpers.go:23]: maybe `vectors_hnsw_pq` to be consistent with naming of flat indexes?

File: adapters/repos/db/file_structure_migration.go [hunk 3]
```diff
@@ -118,3 +132,78 @@ func makeNewRelPath(oldPath, prefix string) (newPath string) {
 	}
 	return
 }
+
+func (db *DB) migratePQVectors(plan migrationPlan) error {
+	for _, parts := range plan {
+		oldClassPath := path.Join(
+			db.config.RootPath,
+			toTitleCase(parts[0].index),
```
Comment (aliszka) [inline adapters/repos/db/file_structure_migration.go:140]: Not sure if I understand it correctly - is this attempt of getting back original className based on lowercased value? If so I'm afraid that may be failing point.

Given the `classname`, original name might have been `ClassName`, `Classname`, `CLASSNAME`, `ClAsSnAmE` etc.
Maybe we could try to find old paths other way round - check which of existing directories matches (after lowercasing) known indexes, and then move files in old path to new path.

File: adapters/repos/db/file_structure_migration.go [hunk 4]
```diff
@@ -118,3 +132,78 @@ func makeNewRelPath(oldPath, prefix string) (newPath string) {
...
+		if err != nil {
+			return fmt.Errorf("check store exists: %w", err)
+		}
+
+		if !exists {
+			continue
+		}
+
+		newStorePath := path.Join(
+			oldClassPath,
```
Comment (aliszka) [inline adapters/repos/db/file_structure_migration.go:158]: if old path is used to build a new one, does it mean that if old one had class/index name with uppercases, new one will also have uppercases? shouldn't be lowercased?

**Comment by antas-marcin** [comment]
@parkerduckworth can we close this PR? as the original [PR](https://github.com/weaviate/weaviate/pull/3765) was already merged into `stable/v1.22` branch and will be ported to `master`

File: adapters/repos/db/file_structure_migration.go (hunk 4 above)
Comment (parkerduckworth) [reply adapters/repos/db/file_structure_migration.go]: You're right, a different approach is needed. Latest commit handles PQ migration differently

**Review by aliszka — APPROVED** [review]

**Pull request merged**